src/core/handle.ml

```diff
@@ -22,12 +22,10 @@ let check_builtin_nat : popt -> sym StrMap.t -> string -> sym -> unit
   match s with
   | "+1" ->
      let builtin = Sign.builtin pos builtins in
-     let symb_0 = builtin "0" in
-     let typ_0 = !(symb_0.sym_type) in
-     let x = Bindlib.new_var mkfree "_" in
-     let typ_s = Ctxt.to_prod [(x, typ_0)] typ_0 in
-     if not (Basics.eq typ_s !(sym.sym_type)) then
-       fatal pos "The type of [%s] is not of the form [%a]"
+     let typ_0 = !((builtin "0").sym_type) in
+     let typ_s = Basics.imply typ_0 typ_0 in
```

Comment:
Actually `imply` is only used here, so I would remove `imply` and use instead:
```ocaml
let typ_s = Bindlib.unbox (_Impl (lift typ_0) (lift typ_0)) in
```
You could share the `lift typ_0` but I would not even bother.

Reply:
ok with sharing (it's really nothing to do):
```
     let typ_0 = lift (!((builtin "0").sym_type)) in
     let typ_s = Bindlib.unbox (_Impl typ_0 typ_0) in
```